Research/ResearchUI/iOS/Step View Controllers/RSDScrollingOverviewStepViewController.swift

```diff
@@ -141,58 +139,279 @@ open class RSDScrollingOverviewStepViewController: RSDOverviewStepViewController
+
+        let cell = self.iconCollectionView.dequeueReusableCell(withReuseIdentifier: iconCollectionViewCellResuableCellId, for: indexPath)
+        
+        // The grid layout stores items as (section, row),
+        // so make sure we use the grid layout to get the correct item index.
+        let itemIndex = self.iconCollectionView.gridLayout.itemIndex(for: indexPath)
+
+        if let overviewCell = cell as? RSDOverviewCollectionViewCell {
+            overviewCell.indexPath = indexPath
+            overviewCell.setDesignSystem(self.designSystem, with: self.backgroundColor(for: .body))
+            if let icons = self.overviewStep?.icons,
+                indexPath.row < icons.count {
+                let icon = icons[itemIndex]
+                overviewCell.imageView?.image = icon.icon?.embeddedImage()
+                overviewCell.titleLabel?.text = icon.title
+                overviewCell.setNeedsUpdateConstraints()
+            }
+        }
+
+        return cell
+    }
+    
+    public func collectionView(_ collectionView: UICollectionView, layout collectionViewLayout: UICollectionViewLayout, referenceSizeForHeaderInSection section: Int) -> CGSize {
+        // Only have a section header in the first section.
+        if section == 0 {
+            return CGSize(width: iconCollectionView.bounds.width, height: self.iconCollectionViewHeaderHeight)
+        }
+        // Otherwise make the size of the section have no height.
+        return CGSize(width: iconCollectionView.bounds.width, height: 0)
+    }
+    
+    public func collectionView(_ collectionView: UICollectionView, viewForSupplementaryElementOfKind kind: String, at indexPath: IndexPath) -> UICollectionReusableView {
+        
+        if kind == UICollectionView.elementKindSectionHeader {
+            let headerCell = collectionView.dequeueReusableSupplementaryView(ofKind: kind, withReuseIdentifier: iconCollectionHeaderTitleCellResuableCellId, for: indexPath)
+
+            if let rsdCell = headerCell as? RSDTitleHeaderCollectionViewHeader {
+                rsdCell.setDesignSystem(self.designSystem, with: self.backgroundColor(for: .header))
+                rsdCell.titleLabel?.text = self.iconCollectionViewHeaderTitle
+            }
+            
+            return headerCell
+        }
+        return UICollectionReusableView()
+    }
+
+    // MARK: Initialization
+
     /// The default nib name to use when instantiating the view controller using `init(step:)`.
     override open class var nibName: String {
         return String(describing: RSDScrollingOverviewStepViewController.self)
     }
-    
+
     /// The default bundle to use when instantiating the view controller using `init(step:)`.
     override open class var bundle: Bundle {
         return Bundle(for: RSDScrollingOverviewStepViewController.self)
     }
+}
+
+/// `RSDTitleHeaderCollectionViewHeader` shows a simple title label.
+public class RSDTitleHeaderCollectionViewHeader: RSDCollectionViewCell {
```

Comment:
If there is a reason for using public (and not open) then this should be marked final.

Reply:
Made it open